fix: PipelineRun cancel-in-progress issue when generateName is used

zakisk opened the pull request:
When PipelineRun is specified using generateName field, PAC unable to filter out and cancel PipelineRuns based on the generateName field due to trailing hyphen in the generateName field, only alphanumeric characters are allowed at begining and end of Kubernetes resources labels thus getting error when filtering PipelineRuns based on originalPRName label. Thus this PR uses originalPRName label instead of annotation to filter the PipelineRun.

https://issues.redhat.com/browse/SRVKP-7337

# Changes <!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

<!-- Describe your changes here- ideally you can get that description straight from your descriptive commit message(s)! -->

# Submitter Checklist

- [x] 📝 Ensure your commit message is clear and informative. Refer to the How to write a git commit message guide. Include the commit message in the PR body rather than linking to an external site (e.g., Jira ticket).

- [x] ♽ Run make test lint before submitting a PR to avoid unnecessary CI processing. Consider installing [pre-commit](https://pre-commit.com/) and running pre-commit install in the repository root for an efficient workflow.

- [x] ✨ We use linters to maintain clean and consistent code. Run make lint before submitting a PR. Some linters offer a --fix mode, executable with make fix-linters (ensure [markdownlint](https://github.com/DavidAnson/markdownlint) and [golangci-lint](https://github.com/golangci/golangci-lint) are installed).

- [ ] 📖 Document any user-facing features or changes in behavior.

- [ ] 🧪 While 100% coverage isn't required, we encourage unit tests for code changes where possible.

- [ ] 🎁 If feasible, add an end-to-end test. See [README](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/README.md) for details.

- [ ] 🔎 Address any CI test flakiness before merging, or provide a valid reason to bypass it (e.g., token rate limitations).

- If adding a provider feature, fill in the following details:

| Git Provider          | Supported |
|-----------------------|-----------|
| GitHub App            | ✅️        |
| GitHub Webhook        | ✅️        |
| Gitea                 | ✅️        |
| GitLab                | ✅️        |
| Bitbucket Cloud       | ✅️        |
| Bitbucket Data Center | ❌️        |

  (update the documentation accordingly)

zakisk commented:
/test go-testing

chmouel commented:
can we have some tests testing this ?

zakisk commented:
you meant E2E?

chmouel commented:
i mean even unit tests as well, you only adjusting the test  but don't actually test the fix here

zakisk committed:
fix: PipelineRun cancel-in-progress issue when generateName is used

When PipelineRun is specified using generateName field, PAC unable
to filter out and cancel PipelineRuns based on the generateName field
due to trailing hyphen in the generateName field, only alphanumeric
characters are allowed at begining and end of Kubernetes resources labels
thus getting error when filtering PipelineRuns based on originalPRName
label. Thus this PR uses originalPRName label instead of annotation to
filter the PipelineRun.

https://issues.redhat.com/browse/SRVKP-7337

Signed-off-by: Zaki Shaikh <zashaikh@redhat.com>

zakisk commented on pkg/pipelineascode/cancel_pipelineruns.go:
```diff
@@ -28,9 +28,9 @@ var cancelMergePatch = map[string]any{
+// cancelAllInProgressBelongingToClosedPullRequest cancels all in-progress PipelineRuns
 // that belong to a specific pull request in the given repository.
-func (p *PacRun) cancelAllInProgressBelongingToPullRequest(ctx context.Context, repo *v1alpha1.Repository) error {
+func (p *PacRun) cancelAllInProgressBelongingToClosedPullRequest(ctx context.Context, repo *v1alpha1.Repository) error {
```
As this cancels PipelineRuns on PR closed, func name seemed bit confusing to me

chmouel commented on pkg/pipelineascode/cancel_pipelineruns_test.go:
```diff
@@ -537,6 +538,54 @@ func TestCancelInProgressMatchingPR(t *testing.T) {
+			cancelledPipelineRuns: map[string]bool{
+				"pr-foo-2": true,
+			},
+			wantLog: "cancel-in-progress: cancelling pipelinerun foo/",
```
is that how the log shows up when generateName is used? it doesn't have the pr name  anymore?

zakisk replied:
got it! will add.

zakisk replied:
it does have it.

chmouel commented:
/lgtm

zakisk commented:
The failed test (`TestGitlabCancelInProgressOnPRClose`) in last E2E run has a flake, I ran it locally 2 time failed, 2 time passed 😄  (failed, passed, passed, failed), re-triggered the job again!

chmouel commented:
it's painful indeed

zakisk commented:
> it's painful indeed

yeah, I tried to solve that in #2036

chmouel commented:
let's not forget to add the backport labels for PR we want to backports
